prefs.js

```diff
@@ -545,6 +548,59 @@ function createPrefsWidgetClass(resource_path, util) {
                     this.monitor_combo.thaw_notify();
                 }
             }
+
+            copy_gnome_terminal_profile() {
+                // Lookup gnome terminal's setting schemas
+                let profile_list_schema, profile_schema;
+                try {
+                    profile_list_schema = util.get_settings_schema('org.gnome.Terminal.ProfilesList');
+                    profile_schema = util.get_settings_schema('org.gnome.Terminal.Legacy.Profile');
+                } catch (e) {
+                    util.show_dialog(this.get_toplevel(), `${e.message} Probably, GNOME Terminal is not installed.`);
+                    return;
+                }
+
+                // Find default gnome terminal profile
+                let profiles_list = Gio.Settings.new_full(profile_list_schema, null, null);
+                let profilePath = profiles_list.settings_schema.get_path();
+                let uuid = profiles_list.get_string('default');
+                let gnome_terminal_profile = Gio.Settings.new_full(
+                    profile_schema,
+                    null,
+                    `${profilePath}:${uuid}/`
+                );
+
+                // Copy color profile
+                try {
+                    const profile_keys = [
+                        'use-theme-colors',
+                        'foreground-color', 'background-color',
+                        'bold-color-same-as-fg', 'bold-color',
+                        'cursor-colors-set', 'cursor-foreground-color', 'cursor-background-color',
+                        'highlight-colors-set', 'highlight-foreground-color', 'highlight-background-color',
+                        'palette',
+                        'bold-is-bright',
+                    ];
+
+                    // Check if key is valid
+                    for (const key of profile_keys) {
+                        const type_gnome_terimnal = util.get_settings_schema_key(profile_schema, key).get_value_type();
+                        const type_ddterm = util.get_settings_schema_key(this.settings.settings_schema, key).get_value_type();
+
+                        if (!type_gnome_terimnal.equal(type_ddterm))
+                            throw new Error(`The type of key '${key}' in GNOME Terminal is not as expected.`);
```

Comment:
Would be nice if the message contained the actual types - see [GLib.VariantType.dup_string()](https://gjs-docs.gnome.org/glib20~2.66.1/glib.varianttype#method-dup_string). This would make debugging easier if something ever goes wrong.